File: financial-domain-starter/src/main/java/com/nixsolutions/financial/discovery/ServiceRecord.java
package com.nixsolutions.financial.discovery;

import javax.validation.constraints.NotNull;
import org.hibernate.validator.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class ServiceRecord
{
  /**
   * Defines the name of a service
   */
  @NotBlank
  private String name;

  /**
   * Defines how to connect to service
   */
  @NotNull
  private DiscoveryType resolution;

  /**
   * Version is not used for now
   */
  private String version;

  /**
   * Self-explaining
   */
  private String port = "8080";
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/discovery/ServiceRecord.kt
package com.nixsolutions.financial.discovery

import org.hibernate.validator.constraints.NotBlank
import javax.validation.constraints.NotNull

class ServiceRecord {
  /**
   * Defines the name of a service
   */
  @NotBlank
  var name: String? = null

  /**
   * Defines how to connect to service
   */
  @NotNull
  var resolution: DiscoveryType? = null

  /**
   * Version is not used for now
   */
  var version: String? = null

  /**
   * Self-explaining
   */
  var port = "8080"
}